pkg/container-utils/docker/docker_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docker
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetContainerImageInfoFromImage(t *testing.T) {
```

Comment:
I added the additional assertions, however I noticed that when pulling a digested image, eg: `docker.io/library/alpine@sha256:e31c3b1cd47718260e1b6163af0a05b3c428dc01fa410baf72ca8b8076e22e72`, `ContainerImageName` is inconsistent between containerd and docker:

containerd: `ContainerImageName:   "docker.io/library/alpine@sha256:e31c3b1cd47718260e1b6163af0a05b3c428dc01fa410baf72ca8b8076e22e72",`
docker: `ContainerImageName:   "docker.io/library/alpine",`

is this intended? should I leave them as-is or change one of them to another format?

Reply:
No, it's not intended and it should be aligned. 

I checked the code and it's because the in Docker we are [parsing](https://github.com/inspektor-gadget/inspektor-gadget/blob/9ff1e81e30f54325a5be2c19ee425d7835b7a290/pkg/container-utils/docker/docker.go#L261) what Docker provides, while for containerd, we are just using what [containerd provides](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/container-utils/containerd/containerd.go#L332). 

Do you want to fix it? If so, please manage it in another PR. This PR could focus only on container digest.